src/Plugins/ComplexCore/src/ComplexCore/Filters/AlignGeometries.cpp

```diff
@@ -0,0 +1,471 @@
+#include "complex/DataStructure/Geometry/VertexGeom.hpp"
+#include "complex/Filter/Actions/CreateArrayAction.hpp"
+#include "complex/Parameters/ChoicesParameter.hpp"
+#include "complex/Parameters/GeometrySelectionParameter.hpp"
+#include "complex/Utilities/DataArrayUtilities.hpp"
+
+#include <string>
+
+namespace
+{
+constexpr complex::int32 k_EMPTY_PARAMETER = -123;
+
+complex::FloatVec3 extractOrigin(const complex::AbstractGeometry& geometry)
+{
+  auto geomType = geometry.getGeomType();
+  switch(geomType)
+  {
+  case complex::AbstractGeometry::Type::Image: {
+    auto& image = dynamic_cast<const complex::ImageGeom&>(geometry);
+    return image.getOrigin();
+  }
+  case complex::AbstractGeometry::Type::RectGrid: {
+    auto& rectGrid = dynamic_cast<const complex::RectGridGeom&>(geometry);
+    const auto& xBounds = rectGrid.getXBounds()->getDataStoreRef();
+    const auto& yBounds = rectGrid.getYBounds()->getDataStoreRef();
+    const auto& zBounds = rectGrid.getZBounds()->getDataStoreRef();
+    complex::FloatVec3 origin(std::numeric_limits<float>::max(), std::numeric_limits<float>::max(), std::numeric_limits<float>::max());
+    for(size_t i = 0; i < xBounds.getNumberOfTuples(); i++)
+    {
+      if(xBounds[i] < origin[0])
+      {
+        origin[0] = xBounds[i];
+      }
+    }
+    for(size_t i = 0; i < yBounds.getNumberOfTuples(); i++)
+    {
+      if(yBounds[i] < origin[1])
+      {
+        origin[1] = yBounds[i];
+      }
+    }
+    for(size_t i = 0; i < zBounds.getNumberOfTuples(); i++)
+    {
+      if(zBounds[i] < origin[2])
+      {
+        origin[2] = zBounds[i];
+      }
+    }
+    return origin;
+  }
+  break;
+  case complex::AbstractGeometry::Type::Vertex: {
+    auto& vertex = dynamic_cast<const complex::VertexGeom&>(geometry);
+    const auto& vertices = vertex.getVertices()->getDataStoreRef();
+    complex::FloatVec3 origin(std::numeric_limits<float>::max(), std::numeric_limits<float>::max(), std::numeric_limits<float>::max());
+
+    for(size_t i = 0; i < vertex.getNumberOfVertices(); i++)
+    {
+      for(size_t j = 0; j < 3; j++)
+      {
+        if(vertices[3 * i + j] < origin[j])
+        {
+          origin[j] = vertices[3 * i + j];
+        }
+      }
+    }
+    return origin;
+  }
+  break;
+  case complex::AbstractGeometry::Type::Edge: {
+    const auto& edge = dynamic_cast<const complex::EdgeGeom&>(geometry);
+    const auto& vertices = edge.getVertices()->getDataStoreRef();
+    complex::FloatVec3 origin(std::numeric_limits<float>::max(), std::numeric_limits<float>::max(), std::numeric_limits<float>::max());
+
+    for(size_t i = 0; i < edge.getNumberOfVertices(); i++)
+    {
+      for(size_t j = 0; j < 3; j++)
+      {
+        if(vertices[3 * i + j] < origin[j])
+        {
+          origin[j] = vertices[3 * i + j];
+        }
+      }
+    }
+    return origin;
+  }
+  break;
+  // 2D
+  case complex::AbstractGeometry::Type::Triangle:
+    [[falthrough]] case complex::AbstractGeometry::Type::Quad:
```

Comment:
```suggestion
    [[fallthrough]];
  case complex::AbstractGeometry::Type::Quad:
```
There's a missing "l" in the attribute. And I'd add a semicolon after the attribute to put it on it's own line. I think it makes it more readable, and it's how the example on cppreference is.